File: library/src/main/java/zjm/cst/dhu/library/utils/DatabaseUtil.java
package zjm.cst.dhu.library.utils;

import org.greenrobot.greendao.query.Query;

import java.util.Arrays;
import java.util.List;
import java.util.Map;

import zjm.cst.dhu.library.Constant;
import zjm.cst.dhu.library.MyApplication;
import zjm.cst.dhu.library.R;
import zjm.cst.dhu.library.mvp.model.NewsChannel;
import zjm.cst.dhu.library.mvp.model.NewsChannelDao;

/**
 * Created by zjm on 2017/5/10.
 */

public class DatabaseUtil {
    public static void setupDatabase() {
        NewsChannelDao newsChannelDao = MyApplication.getNewChannelDao();
        List<String> channelNameList = Arrays.asList(MyApplication.getContext().getResources()
                .getStringArray(R.array.news_channel_list_name));
        List<String> channelIdList = Arrays.asList(MyApplication.getContext().getResources()
                .getStringArray(R.array.news_channel_list_id));
        List<String> channelTypeList = Arrays.asList(MyApplication.getContext().getResources()
                .getStringArray(R.array.news_channel_list_type));
        for (int i = 0; i < channelNameList.size(); i++) {
            String name = channelNameList.get(i);
            String type = channelTypeList.get(2);
            boolean isLike = false;
            if (name.equals(Constant.NEWS_CHANNEL_HEADLINE)) {
                type = channelTypeList.get(0);
            } else if (name.equals(Constant.NEWS_CHANNEL_HOUSE)) {
                type = channelTypeList.get(1);
            }
            if (i < Constant.NEWS_CHANNEL_LIKEINDEX) {
                isLike = true;
            }
            newsChannelDao.insert(new NewsChannel(i, name, channelIdList.get(i), i, type, isLike));
        }
    }

    public static List<NewsChannel> getLikeChannel() {
        Query<NewsChannel> newsChannelQuery = MyApplication.getNewChannelDao().queryBuilder()
                .where(NewsChannelDao.Properties.IsLike.eq(true))
                .orderAsc(NewsChannelDao.Properties.ChannelIndex)
                .build();
        return newsChannelQuery.list();
    }

    public static List<NewsChannel> getUnlikeChannel() {
        Query<NewsChannel> newsChannelQuery = MyApplication.getNewChannelDao().queryBuilder()
                .where(NewsChannelDao.Properties.IsLike.eq(false))
                .orderAsc(NewsChannelDao.Properties.ChannelIndex)
                .build();
        return newsChannelQuery.list();
    }

    public static void updateChannel(Map<Integer, List<NewsChannel>> newsChannelMap) {
        List<NewsChannel> like = newsChannelMap.get(Constant.CHANNEL_LIST_LIKE);
        List<NewsChannel> unlike = newsChannelMap.get(Constant.CHANNEL_LIST_UNLIKE);
        NewsChannel newsChannel;
        for (int i = 0; i < like.size(); i++) {
            newsChannel = like.get(i);
            newsChannel.setIsLike(true);
            updateChannel(newsChannel);
        }
        for (int i = 0; i < unlike.size(); i++) {
            newsChannel = unlike.get(i);
            newsChannel.setIsLike(false);
            updateChannel(newsChannel);
        }
    }


    public static void updateChannel(NewsChannel newsChannel) {
        MyApplication.getNewChannelDao().update(newsChannel);
    }

}

File: library/src/main/kotlin/zjm/cst/dhu/library/utils/data/DatabaseUtil.kt
package zjm.cst.dhu.library.utils.data

import zjm.cst.dhu.library.Constant
import zjm.cst.dhu.library.MyApplication
import zjm.cst.dhu.library.mvp.model.NewsChannel
import zjm.cst.dhu.library.mvp.model.NewsChannelDao
import zjm.cst.dhu.library.R

/**
 * Created by zjm on 2017/5/10.
 */

object DatabaseUtil {

    fun setupDatabase() {
        val newsChannelDao = MyApplication.Companion.newChannelDao
        val channelNameList = MyApplication.Companion.context!!.resources
                .getStringArray(R.array.news_channel_list_name)
        val channelIdList = MyApplication.Companion.context!!.resources
                .getStringArray(R.array.news_channel_list_id)
        val channelTypeList = MyApplication.Companion.context!!.resources
                .getStringArray(R.array.news_channel_list_type)
        for (i in channelNameList.indices) {
            val name = channelNameList[i]
            var type = channelTypeList[2]
            var isLike = false
            if (name == Constant.NEWS_CHANNEL_HEADLINE) {
                type = channelTypeList[0]
            } else if (name == Constant.NEWS_CHANNEL_HOUSE) {
                type = channelTypeList[1]
            }
            if (i < Constant.NEWS_CHANNEL_LIKEINDEX) {
                isLike = true
            }
            newsChannelDao.insert(NewsChannel(i.toLong(), name, channelIdList[i], i, type, isLike))
        }
    }

    val likeChannel: List<NewsChannel>?
        get() {
            val newsChannelQuery = MyApplication.Companion.newChannelDao.queryBuilder()
                    .where(NewsChannelDao.Properties.IsLike.eq(true))
                    .orderAsc(NewsChannelDao.Properties.ChannelIndex)
                    .build()
            return newsChannelQuery.list()
        }

    val unlikeChannel: List<NewsChannel>?
        get() {
            val newsChannelQuery = MyApplication.newChannelDao.queryBuilder()
                    .where(NewsChannelDao.Properties.IsLike.eq(false))
                    .orderAsc(NewsChannelDao.Properties.ChannelIndex)
                    .build()
            return newsChannelQuery.list()
        }

    fun updateChannel(newsChannelMap: Map<Int, List<NewsChannel>?>) {
        val like = newsChannelMap[Constant.CHANNEL_LIST_LIKE]
        val unlike = newsChannelMap[Constant.CHANNEL_LIST_UNLIKE]
        var newsChannel: NewsChannel
        if (like != null) {
            for (i in like.indices) {
                newsChannel = like.get(i)
                newsChannel.isLike = true
                DatabaseUtil.updateChannel(newsChannel)
            }
        }
        if (unlike != null) {
            for (i in unlike.indices) {
                newsChannel = unlike.get(i)
                newsChannel.isLike = false
                DatabaseUtil.updateChannel(newsChannel)
            }
        }
    }


    fun updateChannel(newsChannel: NewsChannel) {
        MyApplication.Companion.newChannelDao.update(newsChannel)
    }

}
